libflux/flux-core/src/semantic/tests.rs

```diff
@@ -3379,7 +3379,7 @@ fn issue_4051() {
         x = f(r:r)
         "#,
         exp: map![
-            "f" => "(r:{S with a:X, b: int, c: float) => {S with a:X, b: int, c: float}",
+            "f" => "(r:{S with a:X, b: int, c: float}) => {S with a:X, b: int, c: float}",
```

Comment:
How did a typo like this get past CI? I would say "Can we add a test for this?" but _this is the test_, so I'm worried we have an issue here.

Reply:
So a quick google search shows we can tell CircleCI to merge the branch with the parent before tests etc. See https://circleci.com/developer/orbs/orb/circleci/build-tools#usage-merge-with-parent

This could be a simple fix here. We do have an internal system for private repos that runs CI against a rebased commit but that system doesn't work with OSS repos. So perhaps we could try this CircleCI config for a week or so.